### PR TITLE
Add Camel-K Alerting PrometheusRule

### PR DESCRIPTION
**Pull request by gmfrasca**

### CODE REVIEW

File: monitoring/overlays/prod/camelk-alerting-rules.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          severity: critical 
+          datahubalert: "true"
+
+      - alert: Camel-K Operator not Running 
+        annotations:
+          message: Camel-K Operator is down
+          summary: The Camel-K Operator pod is not running.
+        expr: |
+          min(min_over_time(kube_pod_container_status_ready{namespace="dh-prod-camel-k", container="camel-k-operator.*"}[5m])) < 1
+        for: 5m
```
Comment (anishasthana) [inline monitoring/overlays/prod/camelk-alerting-rules.yaml:26]: I feel like we can get by without the operator pod being up for a larger period of time. The pod being down just means that new integrations and stuff won't be created right?

File: monitoring/overlays/prod/camelk-alerting-rules.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          min(min_over_time(kube_pod_container_status_ready{namespace="dh-prod-camel-k", container="integration"}[5m])) < 1
+        for: 5m
+        labels:
+          severity: critical 
+          datahubalert: "true"
+
+      - alert: Kafka2Trino Camel-K Integration Pod not Running
+        annotations:
+          message: Kafka2Trino Camel-K Integration is down
+          summary: The pod that runs the Kafka2Trino ETL Integration has not been running for at least 5 minutes.
```
Comment (anishasthana) [inline monitoring/overlays/prod/camelk-alerting-rules.yaml:45]: We will eventually have multiple integrations right? One per topic. How does that map to this alert?

**Review by anishasthana — COMMENTED** [review]
Left some comments

File: monitoring/overlays/prod/camelk-alerting-rules.yaml (hunk 2 above)
Comment (gmfrasca) [reply monitoring/overlays/prod/camelk-alerting-rules.yaml:45]: i suppose we can make the alert text reflect that, ie `There are no Kafka2Trino Integrations currently active`.  I'm not sure if there is a good way to handle an expected number of Integrations scalably (adding/removing a topic would make the alert invalid without also adjusting this definition)

File: monitoring/overlays/prod/camelk-alerting-rules.yaml (hunk 1 above)
Comment (gmfrasca) [reply monitoring/overlays/prod/camelk-alerting-rules.yaml:26]: yeah, we can adjust the fault tolerence here. perhaps an hour?  more than that suggests we actually do have an issue that needs to be addressed

Comment (anishasthana) [reply monitoring/overlays/prod/camelk-alerting-rules.yaml:26]: Agreed. let's do an hour here

Comment (anishasthana) [reply monitoring/overlays/prod/camelk-alerting-rules.yaml:26]: This still needs an update right?

File: monitoring/overlays/prod/camelk-alerting-rules.yaml (hunk 2 above)
Comment (gmfrasca) [reply monitoring/overlays/prod/camelk-alerting-rules.yaml]: I updated this to be essentially a tolerance of 30% down (ex: current < 0.70 * avg)

File: monitoring/overlays/prod/camelk-alerting-rules.yaml (hunk 1 above)
Comment (gmfrasca) [reply monitoring/overlays/prod/camelk-alerting-rules.yaml:26]: dont think so, the 5m count should still be applicable (if something has been down for an hour, and that condition has been true for at least 5 mins, then alert)

**Review by anishasthana — COMMENTED** [review]
/lgtm

**Pull request closed without merging**